vizro-core/src/vizro/models/_dashboard.py

```diff
@@ -32,8 +32,13 @@
 
 
 def _all_hidden(components: List[Component]):
-    """Returns True if all `components` are either None and/or have hidden=True."""
-    return all(component is None or getattr(component, "hidden", False) for component in components)
+    """Returns True if all `components` are either None and/or have hidden=True and/or className contains `d-none`."""
+    return all(
+        component is None
+        or getattr(component, "hidden", False)
+        or "d-none" in getattr(component, "className", "d-inline")
+        for component in components
+    )
```

Comment:
@antonymilne - I know we've changed this back and forth 😅 I tried to do everything in this commit so it's easier to see which changes are required: https://github.com/mckinsey/vizro/pull/393/commits/74031d621cf2201fda4709f51cfa945e3ba54eab
Essentially the `dbc.Nav` / `dbc.Navbar` do not have a `hidden` argument, so we have to go back to one of the previous approaches of providing a `className` for these. Note I am using the classNames from Bootstrap utilities: https://getbootstrap.com/docs/5.0/utilities/display/. Returning None wouldn't work in this case, as we need to access the the html elements `nav-bar` and ´nav-panel´ via their ID later on.

Reply:
This looks good, thanks for making the change so clear just in one commit! 🚀 

Just one thing I'm curious about: why do we put `className="d-none invisible"` and not just `d-none`?